File: WeChatFlight/Flight/src/main/java/com/flight/activity/MainFlight.java
package com.flight.activity;

public class MainFlight {
	
	public static MainFlight getInstance(){
		return MainFlightHolder.sInstance;
	}
	
	// 静态内部类
	private static class MainFlightHolder{
		private static MainFlight sInstance = new MainFlight(); 
	}
	
	private MainFlight(){
		
	}
	
	private OnGameOverCallback gameOverCallback;
	public void setOnGameOverCallback(OnGameOverCallback callback){
		this.gameOverCallback = callback;
	}
	public void setGameOverCallback(long score){
		if (null != gameOverCallback) {
			gameOverCallback.result(score);
		}
	}
	public interface OnGameOverCallback{
		void result(long score);
	}
}

File: WeChatFlight/Flight/src/main/java/com/flight/activity/MainFlight.kt
package com.flight.activity

class MainFlight private constructor() {
    // 静态内部类
    private object MainFlightHolder {
        val sInstance = MainFlight()
    }

    private var gameOverCallback: OnGameOverCallback? = null
    fun setOnGameOverCallback(callback: OnGameOverCallback?) {
        gameOverCallback = callback
    }

    fun setGameOverCallback(score: Long) {
        if (null != gameOverCallback) {
            gameOverCallback!!.result(score)
        }
    }

    interface OnGameOverCallback {
        fun result(score: Long)
    }

    companion object {
        val instance: MainFlight
            get() = MainFlightHolder.sInstance
    }
}